runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -89,7 +90,13 @@ public ILogData load(Long aLong) throws Exception {
      */
     public void write(IToken token, Object data)
         throws OverwriteException {
-        final ILogData ld = new LogData(data);
+        final ILogData ld;
+        if (data instanceof CheckpointEntry) {
+            ld = new LogData(DataType.CHECKPOINT, data);
```

Comment:
Yeah, this is not good, we can't create a new DataType for every Entry type. Make checkpoint a metadata field, and we can check the metadata field.